src/routes/ModuleItem.ts

```diff
@@ -96,60 +75,239 @@ const saveNewModuleItem = async (newModuleItem: ModuleItem, res: Response, newFi
   }
 };
 
-export const registerModuleItem = (req: Request, res: Response) => {
+export const registerModuleItem = async (req: Request, res: Response) => {
   const data: RegisterModuleItem = req.body;
   data.moduleId = req.params.moduleId;
   const newModuleItem: ModuleItem = createModuleItem(data);
+  let code = 200;
   if (data.downloadableFile != null) {
-    const newFile: File = createFile(data, newModuleItem);
-    saveNewModuleItem(newModuleItem, res, newFile);
+    const file: RegisterFile = data.downloadableFile;
+    file.attachedAt = newModuleItem;
+    newModuleItem.hasDownloadableFile = true;
+    code = await registerFile(file);
+  }
+  if (code === 200) {
+    await saveNewModuleItem(newModuleItem, res);
   } else {
-    saveNewModuleItem(newModuleItem, res);
+    res.sendStatus(403);
   }
 };
 
-// link, content, visibility, hasfileupload
-export const changeModuleItem = (req: Request, res: Response) => {
-  const data: ChangeModuleItem = req.body;
-  const { moduleId } = req.params;
+/**
+ * @exports
+ * @async
+ * Updates a moduleItem with the data given by the HTTP-Request
+ * @param {Request} req - Holds the data from the HTTP-Request
+ * @param {Response} res - Used to form the response
+ */
+export const changeModuleItem = async (req: Request, res: Response) => {
+  try {
+    const data: ChangeModuleItem = req.body;
+    const { moduleId } = req.params;
+    const { moduleItemId } = req.params;
+    const moduleItem: any = await getOneObject({
+      where: { id: moduleItemId, moduleId },
+    }, ModuleItem);
+    if (data.content != null) {
```

Comment:
Please use either
`Object.assign(moduleItem, data)`

or 

`for (var attrname in data) { moduleItem[attrname] = data[attrname]; }`

See https://stackoverflow.com/a/171256/11688567